chalk-solve/src/infer/unify.rs

```diff
@@ -464,25 +464,22 @@ impl<'t, I: Interner> Unifier<'t, I> {
             }
 
             (&ConstValue::Unevaluated(ref expr1), &ConstValue::Unevaluated(ref expr2)) => {
```

Comment:
This isn't completely necessary, but what you could do (and in other places with this pattern) is something similar to the `normalize_shallow` we do 
```
let a_val = a_val.try_eval(a_ty, interner);
let b_val = b_val.try_eval(a_ty, interner);
```
which would just add a new function of `ConstValue` `try_eval`. Then this would deduplicate/cleanup this logic a bit.